remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java

```diff
@@ -188,24 +195,19 @@ public CollectionWithPagingInfo<Tag> getTags(StoreRef storeRef, Parameters param
 
     public NodeRef validateTag(String tagId)
     {
-    	NodeRef tagNodeRef = nodes.validateNode(tagId);
-    	if(tagNodeRef == null)
-    	{
-    		throw new EntityNotFoundException(tagId);
-    	}
-    	return tagNodeRef;
+        NodeRef tagNodeRef = nodes.validateNode(tagId);
+        return validateTag(tagNodeRef.getStoreRef(), tagId);
     }
     
     public NodeRef validateTag(StoreRef storeRef, String tagId)
     {
-    	NodeRef tagNodeRef = nodes.validateNode(storeRef, tagId);
-    	if(tagNodeRef == null)
-    	{
-    		throw new EntityNotFoundException(tagId);
-    	}
-    	return tagNodeRef;
+        NodeRef tagNodeRef = nodes.validateNode(storeRef,tagId);
+        if (tagNodeRef == null || nodeService.hasAspect(tagNodeRef, ContentModel.ASPECT_TAGGABLE))
```

Comment:
I don't understand what the aim behind this change is.

Reply:
In this change, we are checking cm:taggable aspect to decide whether a node is tag or not.

Reply:
I'm also confused with this check - you are telling here: if node doesn't exist or if it is a tag then throw an exception. You probably forgot about the NOT part.

Reply:
As per QA testing, it's providing correct result. Please let us know the scenario if it's not working correctly

Reply:
Ohh, now I understand it - or at least I think so - you are checking here if node exists or if it's the root tag, am I right?

Reply:
As mentioned above, In this change we are checking cm:taggable aspect to decide whether a node is tag or not.

Reply:
Ok @pksingh41, so if a node has this `cm:taggable` aspect then it is not a tag, yes?

Reply:
Can you please help us with the scenario where it may fail. We will try to incorporate the required fix.

Reply:
Can you please explain the logic you're trying to write here. We will then be able to help tell you if this really does what you think it does.

Reply:
In this change we are checking cm:taggable aspect to decide whether a node is tag or not. Please suggest if this is not a correct approach.

Reply:
...and if it is a tag then you throw an exception? Or if it's not a tag then you throw an exception? Please can you explain a bit more thoroughly what the if statement does.

Reply:
Only a node which is not a tag can have cm:taggable aspect So we are throwing exception only when a node is not a tag
and that's the validation we are making here.

Reply:
I think that a node which is not a tag and is not tagged yet (simple file for example) could successfully pass this verification - it is a valid node and has not `cm:taggable` aspect.

Reply:
But maybe I'm going to deep into it. In general you are right @pksingh41 that tags shouldn't have this aspect present.

Reply:
> Only a node which is not a tag can have cm:taggable aspect

This doesn't sound like the right logic though. You care whether the node is a tag or not.

This is like:
* I want to check if something is edible
* I know that anything made of concrete is not edible
* The logic for something being edible: `if (object != null || object.isMadeOfConcrete()) { throw NotEdibleException(); }`

Reply:
Can you please suggest the correct approach to validate the tag Id. We will make the required changes.